cmd/sonobuoy/app/args.go

```diff
@@ -22,38 +22,88 @@ import (
 
 	ops "github.com/heptio/sonobuoy/pkg/client"
 	"github.com/heptio/sonobuoy/pkg/config"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
-// TODO (timothysc) - add a general override for --config for all commands which is the sonobuoy config
+const (
+	e2eFocusFlag = "e2e-focus"
+	e2eSkipFlag  = "e2e-skip"
+)
 
-// AddNamespaceFlag initialises a namespace flag
+// AddNamespaceFlag initialises a namespace flag.
 func AddNamespaceFlag(str *string, cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVarP(
 		str, "namespace", "n", config.DefaultPluginNamespace,
 		"The namespace to run Sonobuoy in. Only one Sonobuoy run can exist per namespace simultaneously.",
 	)
 }
 
-// AddE2EModeFlag initialises a mode flag
-func AddE2EModeFlag(mode *ops.Mode, cmd *cobra.Command) {
+// AddModeFlag initialises a mode flag.
+func AddModeFlag(mode *ops.Mode, cmd *cobra.Command) {
 	*mode = ops.Conformance // default
 	cmd.PersistentFlags().Var(
-		mode, "e2e-mode",
+		mode, "mode",
 		fmt.Sprintf("What mode to run sonobuoy in. [%s]", strings.Join(ops.GetModes(), ", ")),
 	)
 }
 
-// AddSonobuoyImage initialises an image url flag
+// AddSonobuoyImage initialises an image url flag.
 func AddSonobuoyImage(image *string, cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(
 		image, "sonobuoy-image", config.DefaultImage,
 		"Container image override for the sonobuoy worker and container",
 	)
 }
 
-// AddKubeconfigFlag adds a kubeconfig flag to the provided command
+// AddKubeconfigFlag adds a kubeconfig flag to the provided command.
 func AddKubeconfigFlag(cfg *Kubeconfig, cmd *cobra.Command) {
 	// The default is the empty string (look in the environment)
 	cmd.PersistentFlags().Var(cfg, "kubeconfig", "Explict kubeconfig file")
+	cmd.MarkFlagFilename("kubeconfig")
+}
+
+// AddSonobuoyConfigFlag adds a SonobuoyConfig flag to the provided command.
+func AddSonobuoyConfigFlag(cfg *SonobuoyConfig, cmd *cobra.Command) {
+	cmd.PersistentFlags().Var(
+		cfg, "config",
+		"path to a sonobuoy configuration JSON file. Overrides --mode",
+	)
+	cmd.MarkFlagFilename("config", "json")
+}
+
+// AddE2EConfig adds two arguments: --e2e-focus and --e2e-skip. These are not taken as pointers, as they are only used by GetE2EConfig.
+func AddE2EConfig(cmd *cobra.Command) {
+	cmd.PersistentFlags().String(
+		e2eFocusFlag, "",
```

Comment:
We should probably present the defaults.

Reply:
What do you mean? The defaults are from Mode.

Reply:
From a UX perspective displaying the default is always helpful to see from the CLI. 

I know it percolates from mode, but we should raise it up to the users view.

Reply:
kk, I'll fill that in.